### ISSUE TITLE
Pointer constraints being immediately removed after creation on native Wayland windows

### ISSUE THREAD

**Issue by Sorixelle** [hyprwm/Hyprland#861]
When a Wayland-native application (in my case, [Looking Glass](https://looking-glass.io/)) attempts to constrain the mouse, the constraint seems to be removed immediately on moving the mouse. Hyprland logs while attempting to reproduce this show the following:
```
[LOG] New mouse constraint at 25006c0
[LOG] Registered signal for owner 25067d0: 25007c8 -> 2506848 (owner: Constraint)
[LOG] Registered signal for owner 25067d0: 25007b8 -> 25067e0 (owner: Constraint)
[LOG] Constraint rechecked: 0, 0 to 1169, 609 for 2450050 (window name: )
[LOG] Registered signal for owner 1e584e0: 2450310 -> 1e58538 (owner: Mouse constraint commit)
[LOG] Constrained mouse to 25006c0
<move mouse here>
[LOG] Callback 1e58560 -> 1e58558, Mouse constraint commit removed.
```

Based on that final log message, it appears to me as if [this call to `unconstrainMouse`](https://github.com/hyprwm/Hyprland/blob/main/src/managers/input/InputManager.cpp#L78) is being fired before it should, especially since it isn't accompanied by a `[LOG] Unconstrained mouse from <id>` message that I observed when testing similar functionality in Minecraft, which uses XWayland.

Reproduction might be tricky for this one, as Looking Glass requires a whole host of complicated VM and PCI passthrough setup to get working, and I'm unaware of any other Wayland-native apps with similar mouse capturing behaviour. although according to #536, SDL applications seem to be exhibiting similar behaviour, so that could be another way to test this. I'm unaware of any SDL apps that try to grab the pointer off the top of my head though. In any case, reproduction should look something like:

1. Open a Wayland-native application that tries to grab the cursor
2. Try to move the mouse outside of the window

Expected: The mouse cannot be moved outside the window.
Actual: The mouse moves outside the window.

**Comment by vaxerski** [comment]
fixed in 83e4006b16e79e747888c507243c37eefabe9790

**Comment by Sorixelle** [comment]
Tested, can confirm that commit fixes this issue. Thanks!
